docs/proposals/cluster-authorization/README.md

```diff
@@ -0,0 +1,281 @@
+---
+title: Cluster(karmada-agent) Authorization
+authors:
+- "@everpeace"
+reviewers:
+- T.B.D.
+approvers:
+- T.B.D.
+
+creation-date: 2024-01-29
+update-date: 2024-01-29
+
+---
+
+# Cluster(karmada-agent) Authorization
+
+## Summary
+
+_Cluster Authorization_ allows karmada-agents(member clusters) permissions to be managed more strictly. By enabling this feature, a karmada-agents will be restricted to the objects it is associated with. For example, a karmada-agent can only access the Secret referenced by the Cluster to which it is associated. This minimizes the security risk of a karmada-agent's credentials being compromised.
+
+
+## Motivation
+
+As of karmada v1.8, karmada-agent uses `user: system:node:<cluster-name>, groups: [system:nodes]` identity when accessing to the karmada control plane. This is because it relies on the Kubernetes auto approving controller to ensure long-term, secure and stable access from karmada-agent to the control plane (ref: https://kubernetes.io/docs/reference/access-authn-authz/certificate-signing-requests/#kubernetes-signers).
+
+Since the `system:nodes` group has strong cluster-wide privileges([ClusterRole and ClusterRoleBinding](https://github.com/karmada-io/karmada/blob/v1.8.0/charts/karmada/templates/_karmada_bootstrap_token_configuration.tpl#L87-L201)), any karmada-agent can have unrestricted access to `Cluster, Lease, Secret`, etc. So any karmada-agent can modifies/deletes critical resources owned by other karmada-agents.
+
+Moreover, because any agents registers member cluster's token and impersonate token in control plane by default, any karmada-agents can invoke impersonate access to any member clusters if network is reachable.  It's dangerous when member clusters are managed by other teams/organizations from control plane. In this situation, whole clusters(all the members, and control plane) would be taken by malicious attacker even if only one member cluster is compromised.
+
+Thus, karmada should introduce a way such that karmada-agent can run with the least privilege.
+
+## Proposal
+
+Similar to [Kubernetes Node Authorization](https://kubernetes.io/docs/reference/access-authn-authz/node/), Cluster Authorization consists of two components:
+
+- Mutating Admission Webhook:
+  - when a karmada-agent(`system:node:member-1`) creates some resources in control plane, this webhook puts `clusterrestriction.karmada.io/owner: "system:node:member-1"`
```

Comment:
Does it mean to add this information to the annotations of the resource?

Reply:
Yes. That's right. The admission webhook puts the annotation to all the resources created by karmada-agent.

I will update the sentence to be more precise.

```suggestion
  - when a karmada-agent(`system:node:member-1`) creates some resources in control plane, this webhook puts `clusterrestriction.karmada.io/owner: "system:node:member-1"` annotaion to the resources
```